### PR TITLE
New Entry: C++ Math Functions: `log()`

### LINKED ISSUE #956: [Term Entry] C++ Math Functions: `log()`
### Reason/inspiration (optional)

Under `cpp/concepts/math-functions/terms/`, create a new term entry for the `log()` function.

- In the `terms/` folder, make a new folder called `log`
- Inside the `log` folder, create a new markdown file called `log.md`
- Inside the file, be sure to include a "Syntax" section with a \```pseudo code snippet.
- Let's include an "Example" section with a \```cpp code snippet.
- Lastly, let's include a "Codebyte Example" with a runnable \```codebyte/cpp

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] I agree to follow Codecademy Doc's Code of Conduct

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by Hitarth-Kothari**
### Description

Closes #956 

Created folder named log in path : `cpp/concepts/math-functions/terms/`
Created a file named log.md, which includes all requirements for a Term Entry Issue

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/cpp/concepts/math-functions/terms/log/log.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+```codebyte/cpp
+#include <iostream>
+#include <math.h>
+int main() {
+  double x = 10;
+  double result;
+  result = log(x);
+  std::cout << "The log of " << x << " is " << result << "\n";
+}
```
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/log/log.md:48]:
```suggestion
#include <iostream>
#include <math.h>

int main() {
  double x = 10;
  double result;
  result = log(x);
  std::cout << "The log of " << x << " is " << result << "\n";
}
```

File: content/cpp/concepts/math-functions/terms/log/log.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
+---
+Title: 'log()'
+Description: 'Returns the natural log of the argument.'
```
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/log/log.md:3]:
```suggestion
Description: 'Returns the natural, base-e logarithm of the argument.'
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@Hitarth-Kothari Finished with my review and left you some comments/suggestions. 😄

**Comment by Hitarth-Kothari** [comment]
@Dusch4593 I tried running yarn to fix some of these issues, but I keep getting an error
Invalid configuration file `content\cpp\concepts\math-functions\terms\sqrt\sqrt.md`: Cannot find module '@codecademy/prettier-config'
Any idea how to fix?

**Comment by Dusch4593** [comment]
@Hitarth-Kothari The GH checks to all be passing. But I'm wondering still how you got that error? Especially because it's complaining about a different file pertaining to a different function other than `log()`. What `yarn` command did you run?

**Comment by Hitarth-Kothari** [comment]
Yarn format <path to file>

**Comment by Dusch4593** [comment]
> Yarn format

I think if meant to format all files, it's `yarn format:all`. For a specific file, it's `yarn format path/to/markdown/file.md`.

However, usually when you `commit` new changes in your branch, we have a [husky hook](https://github.com/Codecademy/docs/blob/main/.husky/pre-commit) that formats/lints all relevant files for you prior to `push`ing back up to the PR. 😄

**Comment by Hitarth-Kothari** [comment]
Tbh I tried both for the isfinite file, and i kept getting the same error, and i dont know if the isfinite will pass the workflow tests since I was unable to run the yarn code

**Comment by Dusch4593** [comment]
> Tbh I tried both for the isfinite file, and i kept getting the same error, and i dont know if the isfinite will pass the workflow tests since I was unable to run the yarn code

I just tried out on my end and ran:

```
yarn format content/cpp/concepts/math-functions/terms/log/log.md
```
It formatted as expected. This command is what's giving you the error?

File: content/cpp/concepts/math-functions/terms/log/log.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  double x = 5;
+  double result;
+  result = log(x);
+  std::cout << "The log of " << x << " is " << result << "\n";
+  // Output: The log of 5 is 1.60944
+}
+```
+
+## Codebyte Example
+
```
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/log/log.md:43]:
```suggestion

The following example is runnable and returns the natural log of `10`:

```

**Review by Dusch4593 — APPROVED** [review]
@Hitarth-Kothari formatting aside, this PR looks good from my end. 😄 I left one last suggestion but am approving for the next review.

**Comment by Hitarth-Kothari** [comment]
I just retried running the command, it still doent work
I think I'm missing a file name preetier-config

**Comment by Dusch4593** [comment]
> @Dusch4593 I just retried running the command, it still doent work I think I'm missing a file name preetier-config

Hmm 🤔 And you're still getting the error even after running with forward slashes `/`?

`yarn format content/cpp/concepts/math-functions/terms/log/log.md`

**Comment by Dusch4593** [comment]
Would you be able to post a screenshot of the error you got on the command line?

**Comment by Hitarth-Kothari** [comment]
Screenshot (31)

**Comment by Hitarth-Kothari** [comment]
Screenshot (32)

**Comment by Dusch4593** [comment]
@Hitarth-Kothari Hmm 🤔 When you're editing entries locally, such as `log.md` in this case, are you doing so from a code editor like VS Code?

**Comment by Hitarth-Kothari** [comment]
I've been using VScode

File: content/cpp/concepts/math-functions/terms/log/log.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+The following example is runnable and returns the natural log of `10`:
```
Comment (SSwiniarski) [inline content/cpp/concepts/math-functions/terms/log/log.md:44]:
```suggestion
The following example is runnable and returns the natural log of `10`:

```

**Review by SSwiniarski — COMMENTED** [review]
@Hitarth-Kothari here's an edit I think will fix the test failure.

**Comment by Hitarth-Kothari** [comment]
Thanks for the suggestions. I am unable to run yarn for some reason and just cant find the formatting issues

**Comment by Dusch4593** [comment]
@Hitarth-Kothari no worries 😄 I went ahead and fixed the formatting with ae2711556977abd09d557851e0570101ee8a1ca9 to get the checks to pass. I'm sorry for what's going through with `yarn` on your end. I want you to know that I've pinged my team about this and want to learn more about what caused the error and how to navigate it next time.

For now, though, @SSwiniarski do you think this now looks good to merge?

**Review by SSwiniarski — APPROVED** [review]
@Dusch4593 Looks Good.

**Comment by Dusch4593** [comment]
@Hitarth-Kothari This looks good to merge! 👍🏻 I made some last-minute, light edits in 7894ea13e53bc9b8f7fb8fb2ff3ed528382e478f and c13baaef263e595dc3dd9aa93735737e29f0320f.

**Comment by Dusch4593** [comment]
@Hitarth-Kothari Congrats, your contribution is now live! 🎉 🎊 🥳

Live entry link here: https://www.codecademy.com/resources/docs/cpp/math-functions/log

**Pull request merged**